#! /bin/sh /usr/share/dpatch/dpatch-run
## 212-print-test-results.dpatch by  <kmccarty@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Explicitly output test program exit codes.

@DPATCH@
diff -urNad cernlib-2006.dfsg.2~/src/config/Imake.rules cernlib-2006.dfsg.2/src/config/Imake.rules
--- cernlib-2006.dfsg.2~/src/config/Imake.rules	2005-04-18 15:39:25.000000000 +0000
+++ cernlib-2006.dfsg.2/src/config/Imake.rules	2007-12-13 18:52:12.000000000 +0000
@@ -2373,13 +2373,15 @@
 test:: ProgramTargetName(program) data inputdata			@@\
 	@echo Testing $(PACKAGE_NAME)					@@\
 	@echo "  ... using" program						@@\
-	@Concat(.,$(PATHSEP))ProgramTargetName(program)			@@\
+	@Concat(.,$(PATHSEP))ProgramTargetName(program)	; \		@@\
+	result="$$?" ; echo "Test result = $$result" ; exit "$$result"	@@\
 									@@\
 else									@@\
 test:: ProgramTargetName(program) data inputdata			@@\
 	@echo Testing $(PACKAGE_NAME)					@@\
 	@echo "  ... using" program					@@\
-	@Concat(.,$(PATHSEP))ProgramTargetName(program) < inputdata	@@\
+	@Concat(.,$(PATHSEP))ProgramTargetName(program) < inputdata ; \	@@\
+	result="$$?" ; echo "Test result = $$result" ; exit "$$result"	@@\
 									@@\
 endif									@@\
 									@@\
